test/libsolidity/util/TestFileParser.h

```diff
@@ -308,16 +309,38 @@ class TestFileParser
 		using TokenDesc = std::pair<Token, std::string>;
 
 		/// Advances current position in the input stream.
-		void advance() { ++m_char; }
-		/// Returns the current character.
-		char current() const { return *m_char; }
-		/// Peeks the next character.
-		char peek() const { auto it = m_char; return *(it + 1); }
+		void advance() {
+			solAssert(m_char != m_line.end(), "Cannot advance beyond end.");
+			++m_char;
+		}
+
+		/// Returns the current character or EOS if at end of input.
+		char current() const {
+			if (m_char == m_line.end())
+				return '\0';
+
+			return *m_char;
+		}
+
+		/// Peeks the next character or EOS if that would be at (or beyond) the end of input.
+		char peek() const {
+			if (std::distance(m_char, m_line.end()) < 2)
+				return '\0';
+
+#if __cplusplus >= 201703L
+			return std::next(m_char, 1);
+#else
+			auto next = m_char;
+			std::advance(next, 1);
+			return *next;
+#endif
+		}
+
 		/// Returns true if the end of a line is reached, false otherwise.
 		bool isEndOfLine() const { return m_char == m_line.end(); }
 
 		std::string m_line;
-		std::string::iterator m_char;
+		std::string::const_iterator m_char;
```

Comment:
This change was needed in order to be able to access `std::distance(A, B)` that requires `A` and `B` to be of same type.